core/src/main/java/com/linecorp/armeria/common/HttpResponseWriter.java

```diff
@@ -34,18 +34,18 @@ default void close(AggregatedHttpResponse res) {
         HttpData content = null;
         try {
             requireNonNull(res, "res");
-
             final ResponseHeaders headers = res.headers();
-            final HttpStatus status = headers.status();
             content = res.content();
-            final boolean contentAlwaysEmpty = isContentAlwaysEmptyWithValidation(status, content);
-
             if (!tryWrite(headers)) {
                 return;
             }
 
-            // Add content if not empty.
-            if (!contentAlwaysEmpty && !content.isEmpty()) {
+            if (headers.status().isContentAlwaysEmpty()) {
```

Comment:
Need to release the resource?
```suggestion
            if (headers.status().isContentAlwaysEmpty()) {
                content.close();
```

Reply:
It's released here. 😉 
https://github.com/line/armeria/pull/5862/files/f91510580f795596cb30717b66b423c3d3af9ad1#diff-f288d29e01958b60c9befbce9e62c6aa2da2545dcc4f354047eb3970bd7816adR63

Reply:
I missed that point.